src/drivers/indexeddb.js

```diff
@@ -181,27 +182,164 @@
             }
         }
 
+        // Initialize a singleton container for all running localForages.
+        if (!dbContexts) {
+            dbContexts = {};
+        }
+
+        // Get the current context of the database;
+        var dbContext = dbContexts[dbInfo.name];
+
+        // ...or create a new context.
+        if (!dbContext) {
+            dbContext = {
+                // Running localForages sharing a database.
+                forages: [],
+                // Shared database.
+                db: null
+            };
+            // Register the new context in the global container.
+            dbContexts[dbInfo.name] = dbContext;
+        }
+
+        // Register itself as a running localForage in the current context.
+        dbContext.forages.push(this);
+
+        // Create an array of readiness of the related localForages.
+        var ready = [];
+        for (var j = 0; j < dbContext.forages.length; j++) {
+            var forage = dbContext.forages[j];
+            if (forage !== this) { // Don't wait for itself...
+                ready.push(forage.ready());
+            }
+        }
+
+        // Take a snapshot of the related localForages.
+        var forages = dbContext.forages.slice(0);
+
+        // Initialize the connection process only when
+        // all the related localForages are ready.
+        return Promise.all(ready).then(function() {
+            dbInfo.db = dbContext.db;
+            // Get the connection or open a new one without upgrade.
+            return _getOriginalConnection(dbInfo);
+        }).then(function(db) {
+            dbInfo.db = db;
+            if (_isUpgradeNeeded(dbInfo, self._defaultConfig.version)) {
+                // Reopen the database for upgrading.
+                return _getUpgradedConnection(dbInfo);
+            }
+            return db;
+        }).then(function(db) {
+            dbInfo.db = dbContext.db = db;
+            self._dbInfo = dbInfo;
+            // Share the final connection amongst related localForages.
+            for (var k in forages) {
+                var forage = forages[k];
+                if (forage !== self) { // Self is already up-to-date.
+                    forage._dbInfo.db = dbInfo.db;
+                    forage._dbInfo.version = dbInfo.db;
+                }
+            }
+        });
```

Comment:
My first idea was indeed to wait for other localForages to be ready, and by "ready" I mean "fulfilled" (sorry if my comment in the code wasn't clear enough about that). This goal of this implementation is to act as a kind of "transaction" actually. Thus, if several instance are created on the same database and one of them fails (like in @thgreasi's example) this implementation prevents, indeed, the next ones to be created. The idea behind that is to avoid any side-effect or inconsistency which might result from letting the process continuing. I mean, if multiples stores are involved in the same database, they are highly probably related somehow. And if an error is not explicitly handled at the initialization time, it has to be handled at a later time anyway.

That being said, I see your point. While my solution forces users to handle errors at initialization time, your solution to wait for localForages to be "not pending" (i.e. fulfilled or rejected) is indeed less restrictive and let them decide where and when to handle errors. So, on second thought, you might be right.

However, the way you suggest to achieve this goal can't work. Just adding an empty `catch()` on a promise doesn't change the status of the latter and returns it as is (i.e fulfilled or rejected). So the promise array will be globally fulfilled or rejected in the same way as before. On the other hand, a solution would be to return a resolved promise in both cases with:

```javascript
ready.push(forage.ready().catch(function() {
    return Promise.resolve();
});
```

Whereas this pattern could be confusing, if not commented, it gets the job done.

Now, the question which has to be decided is: what behavior do we want for users?

*Note: I hope my english is clear enough (here or in the code), it's not my native language...*